packages/astro/src/env/vite-plugin-env.ts

```diff
@@ -47,11 +48,7 @@ export function astroEnv({
 				fileURLToPath(settings.config.root),
 				'',
 			);
-			for (const [key, value] of Object.entries(loadedEnv)) {
-				if (value !== undefined) {
-					process.env[key] = value;
-				}
-			}
+			(globalThis as any)[ENV_SYMBOL] ??= loadedEnv;
```

Comment:
@bluwy actually, isn'it going to only set it once?

Reply:
Oh yeah it should be `=` only. Good catch.